### ISSUE TITLE
Support JDK 17 in Flow

### ISSUE THREAD

**Issue by mshabarov** [vaadin/flow#13572]
Make Flow support JDK 17 to unblock the support of Spring Boot 3.
Known issues: 
- `fusion-endpoints` needs JDK 17 to compile after spring upgrade
- Spring 6 and Spring Boot 3 requires JDK 17.

**Comment by Artur-** [comment]
Could you clarify this a bit? Flow supports Java 17 today

**Comment by mshabarov** [comment]
Right. This is not an issue at all. We expect that Flow would require JDK 17 after migrating to Servlet 5 anyway. Thanks @Artur-
